modules/odr_core/odr_core/crud/embedding.py

```diff
@@ -0,0 +1,246 @@
+
+    # embedding: Iterable[ndarray] = list(model.embed(image))
+    # return embedding[0].tolist()
+
+
+def create_content_embedding(db: Session, content_embedding: ContentEmbeddingCreate) -> ContentEmbedding:
+    db_content_embedding = ContentEmbedding(
+        content_id=content_embedding.content_id,
+        embedding=content_embedding.embedding,
+        embedding_engine_id=content_embedding.embedding_engine_id,
+        from_user_id=content_embedding.from_user_id,
+        from_team_id=content_embedding.from_team_id,
+        created_at=datetime.now(timezone.utc)
+    )
+    db.add(db_content_embedding)
+    db.commit()
+    db.refresh(db_content_embedding)
+    return db_content_embedding
+
+
+def update_content_embedding(db: Session, content_embedding_id: int, content_embedding: ContentEmbeddingUpdate) -> Optional[ContentEmbedding]:
+    db_content_embedding = db.query(ContentEmbedding).filter(ContentEmbedding.id == content_embedding_id).first()
+    if db_content_embedding:
+        for key, value in content_embedding.model_dump(exclude_unset=True).items():
+            setattr(db_content_embedding, key, value)
+        db.commit()
+        db.refresh(db_content_embedding)
+    return db_content_embedding
+
+
+def delete_content_embedding(db: Session, content_embedding_id: int) -> bool:
+    db_content_embedding = db.query(ContentEmbedding).filter(ContentEmbedding.id == content_embedding_id).first()
+    if db_content_embedding:
+        db.delete(db_content_embedding)
+        db.commit()
+        return True
+    return False
+
+
+def query_content_embedding(db: Session, query: EmbeddingVectorQuery, skip: int = 0, limit: int = 100) -> List[ContentEmbedding]:
+    if len(query.embedding) != 512:
+        raise ValueError(f"Invalid embedding length, expected 512 got {len(query.embedding)}")
+
+    return db.query(ContentEmbedding) \
+            .filter(ContentEmbedding.embedding_engine_id == query.embedding_engine_id) \
+            .order_by(ContentEmbedding.embedding.l2_distance(query.embedding)) \
+            .offset(skip).limit(limit).all()
+
+
+def create_annotation_embedding(db: Session, annotation_embedding: AnnotationEmbeddingCreate) -> AnnotationEmbedding:
+    db_annotation_embedding = AnnotationEmbedding(
+        annotation_id=annotation_embedding.annotation_id,
+        embedding=annotation_embedding.embedding,
+        embedding_engine_id=annotation_embedding.embedding_engine_id,
+        from_user_id=annotation_embedding.from_user_id,
+        from_team_id=annotation_embedding.from_team_id,
+        created_at=datetime.now(timezone.utc)
+    )
+    db.add(db_annotation_embedding)
+    db.commit()
+    db.refresh(db_annotation_embedding)
+    return db_annotation_embedding
+
+
+def update_annotation_embedding(db: Session, annotation_embedding_id: int, annotation_embedding: AnnotationEmbeddingUpdate) -> Optional[AnnotationEmbedding]:
```

Comment:
What is the thought behind the optional here?

Reply:
Yeah, It doesn't make sense.
Fixed it.